<launch>
  <!-- arguments -->
  <arg name="web3_http_provider" default="http://127.0.0.1:8545" />
  <arg name="ipfs_http_provider" default="http://127.0.0.1:5001" />
  <arg name="lighthouse_contract" default="0x9c8DD1E767f54520B4fb1A6E306F172EA8492f9c" />
  <arg name="enable_generator" default="false" />
  <arg name="enable_signer" default="true" />
  <arg name="eth_account_address" default="" />

  <!-- market group -->
  <group ns="infochan">
    <node pkg="robonomics_lighthouse" type="infochan_node" name="infochan">
      <param name="ipfs_http_provider" value="$(arg ipfs_http_provider)" />
      <param name="lighthouse_contract" value="$(arg lighthouse_contract)" />
    </node>

    <node pkg="robonomics_lighthouse" type="signer_node" name="signer" if="$(arg enable_signer)" output="screen">
      <param name="web3_http_provider" value="$(arg web3_http_provider)" />
      <param name="eth_account_address" value="$(arg eth_account_address)"
             if="$(eval len(arg('eth_account_address')) > 0)"/>
    </node>

    <node pkg="robonomics_lighthouse" type="generator_node" name="generator" if="$(arg enable_generator)" />
  </group>
</launch>
